riocli/deployment/update.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2023 Rapyuta Robotics
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import re
+import sys
+
+import click
+from click_help_colors import HelpColorsCommand
+
+from rapyuta_io import Client, DeploymentPhaseConstants
+
+from riocli.config import new_client
+from riocli.constants import Symbols, Colors
+from riocli.utils import tabulate_data
+from riocli.utils.spinner import with_spinner
+
+@click.command(
+    'update',
+    cls=HelpColorsCommand,
+    help_headers_color=Colors.YELLOW,
+    help_options_color=Colors.GREEN,
+)
+@click.option('--force', '-f', '--silent', is_flag=True, default=False,
+              help='Skip confirmation')
+@click.option('--update-all', '-a', is_flag=True, default=False,
+              help='Updates all deployments')
+@click.argument('deployment-name-or-regex', type=str, default="")
+@with_spinner(text="Updating deployment...")
+def update_deployment(force: bool, deployment_name_or_regex: str, update_all: bool, spinner=None) -> None:
+    """
+    Delete the deployment from the Platform
+    """
+    client = new_client()
+
+    if not(deployment_name_or_regex or update_all):
+        return
+
+    try:
+        deployments = fetch_deployments(client, deployment_name_or_regex, update_all)
+    except Exception as e:
+        spinner.text = click.style(
+            'Failed to update deployment(s): {}'.format(e), Colors.RED)
+        spinner.red.fail(Symbols.ERROR)
+        raise SystemExit(1) from e
+
+    if not deployments:
+        return
+
+    headers = ["name", "guid"]
+    data = [[deployment.name, deployment.deploymentId] for deployment in deployments]
+
+    spinner.write("")
+    tabulate_data(data, headers)
+
+    if not force:
+        with spinner.hidden():
+            click.confirm("Do you want to update above deployment(s)?", default=True, abort=True)
+
+    try:
+
+        for deployment in deployments:
+            dep = client.get_deployment(deployment["deploymentId"])
+            component_context = get_component_context(dep.get("componentInfo", {}))
+            payload = {
+                "service_id": dep["packageId"],
+                "plan_id": dep["planId"],
+                "deployment_id": dep["deploymentId"],
+                "context": {
+                    "component_context": component_context
+                }
+            }
+            client.update_deployment(payload)
```

Comment:
Doing `get_deployment` sequentially might become expensive for multiple deployments. Please check the performance with 10+ deployments. You can multithread this part.

This will improve the overall performance of the command.